tutorials/ingest-real-time-websocket-data.md

```diff
@@ -187,157 +207,128 @@ universal, but you can experiment with different batch sizes
 Using batching is a fairly common pattern when ingesting data into TimescaleDB
 from Kafka, Kinesis, or websocket connections.
 
-Now you can see
-how to implement a batching solution in Python with Psycopg2.
-
-### Implement batching with Psycopg2
-
-Remember to implement the ingestion logic within the `on_event` function that
+You can implement a batching solution in Python with Psycopg2.
+You can implement the ingestion logic within the `on_event` function that
 you can then pass over to the websocket object.
 
 This function needs to:
 
-1.  Check if the item is a data item, and not some websocket metadata.
-1.  Adjust the data so that it fits the database schema, including the data types, and order of columns.
+1.  Check if the item is a data item, and not websocket metadata.
+1.  Adjust the data so that it fits the database schema, including the data
+    types, and order of columns.
 1.  Add it to the in-memory batch, which is a list in Python.
-1.  If the batch reaches a certain size, insert the data and reset or empty the list.
-
-Here's the full implementation:
-
-```python
-from psycopg2.extras import execute_values
-conn = psycopg2.connect(database="tsdb", 
-                        host="host", 
-                        user="tsdbadmin", 
-                        password="passwd",
-                        port="66666")
-columns = ["time", "symbol", "price", "day_volume"]
-current_batch = []
-MAX_BATCH_SIZE = 100
-def _on_event(self, event):
-    if event["event"] == "price":
-        # data record
-        timestamp = datetime.utcfromtimestamp(event["timestamp"])
-        data = (timestamp, event["symbol"], event["price"], event.get("day_volume"))
-
-        # add new data record to batch
-        current_batch.append(data)
-            
-        # ingest data if max batch size is reached then reset the batch
-        if len(current_batch) == MAX_BATCH_SIZE:
-            cursor = conn.cursor()
-            sql = f"""
-            INSERT INTO {DB_TABLE} ({','.join(columns)}) 
-            VALUES %s;"""
-            execute_values(cursor, sql, data)
-            conn.commit()
-            current_batch = []
-```
-
-Make sure you use `execute_values()` or some other Psycopg2 function that
-allows inserting multiple records in one transaction.
-
-After you have implemented the `on_event` function, your Python script can connect to
-the websocket server and ingest data in real time.
+1.  If the batch reaches a certain size, insert the data, and reset or empty the list.
 
-## Full code example
+## Ingesting data in real-time
```

Comment:
This procedure is still broken:

![image](https://github.com/timescale/docs/assets/3914967/20c85ca4-bc5b-4e8f-b6f4-2003d9e1a2f0)